src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -1,164 +1,581 @@
+                        var temp = _factory.SynthesizedLocal(dagTemp.Type, syntax: _factory.Syntax, kind: SynthesizedLocalKind.SwitchCasePatternMatching);
+                        _map.Add(dagTemp, _factory.Local(temp));
+                        _temps.Add(temp);
+                        result = _factory.Local(temp);
                     }
-                default:
-                    return pattern;
-            }
-        }
 
-        private BoundExpression MakeIsConstantPattern(BoundConstantPattern loweredPattern, BoundExpression loweredInput)
-        {
-            return CompareWithConstant(loweredInput, loweredPattern.Value);
-        }
+                    return result;
+                }
 
-        private BoundExpression MakeIsDeclarationPattern(BoundDeclarationPattern loweredPattern, BoundExpression loweredInput)
-        {
-            Debug.Assert(((object)loweredPattern.Variable == null && loweredPattern.VariableAccess.Kind == BoundKind.DiscardExpression) ||
-                         loweredPattern.Variable.GetTypeOrReturnType() == loweredPattern.DeclaredType.Type);
+                public ImmutableArray<LocalSymbol> AllTemps()
+                {
+                    return _temps.ToImmutableArray();
+                }
+
+                public void AssignTemp(BoundDagTemp dagTemp, BoundExpression value)
+                {
+                    _map.Add(dagTemp, value);
+                }
+            }
 
-            if (loweredPattern.IsVar)
+            private bool LowerDecision(BoundDagDecision decision)
             {
-                var result = _factory.Literal(true);
+                var oldSyntax = _factory.Syntax;
+                _factory.Syntax = decision.Syntax;
+                var result = LowerDecisionCore(decision);
+                _factory.Syntax = oldSyntax;
+                return result;
+            }
 
-                if (loweredPattern.VariableAccess.Kind == BoundKind.DiscardExpression)
+            private bool LowerDecisionCore(BoundDagDecision decision)
+            {
+                void addConjunct(ref IsPatternTranslator self, BoundExpression expression)
                 {
-                    return _factory.MakeSequence(loweredInput, result);
+                    // PROTOTYPE(patterns2): could handle constant expressions more efficiently.
+                    if (self._sideEffectBuilder.Count != 0)
+                    {
+                        expression = self._factory.Sequence(ImmutableArray<LocalSymbol>.Empty, self._sideEffectBuilder.ToImmutable(), expression);
+                        self._sideEffectBuilder.Clear();
+                    }
+
+                    self._conjunctBuilder.Add(expression);
                 }
 
-                Debug.Assert((object)loweredPattern.Variable != null && loweredInput.Type.Equals(loweredPattern.Variable.GetTypeOrReturnType(), TypeCompareKind.AllIgnoreOptions));
+                var input = _tempAllocator.GetTemp(decision.Input);
```

Comment:
>var [](start = 16, length = 3)

Please spell out the type.

Reply:
Resolved in [https://github.com/dotnet/roslyn/pull/23357/commits/758fd76932a6d9bcb640504c8fdef57fb5adecec](https://github.com/dotnet/roslyn/pull/23357/commits/758fd76932a6d9bcb640504c8fdef57fb5adecec) of [https://github.com/dotnet/roslyn/pull/23357](https://github.com/dotnet/roslyn/pull/23357)

---
In reply to: [156540610](https://github.com/dotnet/roslyn/pull/23209#discussion_r156540610) [](ancestors = 156540610)